vtex-apps/return-app
# Order list frontend step2

What this PR changes:
Add the Order list and Return details pages to V3.

How to test it:
[this workspace](https://selectorder--powerplanet.myvtex.com/account?__bindingAddress=www.we-demostore.com/es#/my-returns/add)

Order list page:
Use the items in the response to create the order status - MEDIUM;
![1 1](https://user-images.githubusercontent.com/96049132/164193525-424ffa45-ef73-4f87-b9a3-bf41377dc055.png)

Go to order details page when user clicks the button /my-returns/add/:orderId - HIGH;
Order details page:
Handle loading state - HIGH;
On this gif we can see the order details selecting an order and loading state.
![1 2](https://user-images.githubusercontent.com/96049132/164193590-802a6ad1-8896-422f-ab98-b888f6683928.gif)

Handle different errors with custom messages in the order details page - HIGH;
![2 2](https://user-images.githubusercontent.com/96049132/164213972-2eae63ee-19d1-4981-943b-6b4e5cdcb503.png)



Product table - See product and quantity available to be returned - HIGH;
Handle exception cases - MEDIUM:
Show quantity invoiced and items already processed (This way user can understand why quantity to return is different from 
the received one)
![2 3 4](https://user-images.githubusercontent.com/96049132/164194133-8266b592-44ae-4b41-9bbc-8d142cdc930f.png)
![2 3-4](https://user-images.githubusercontent.com/96049132/164194145-5bfcca50-44f0-4cdf-b4e5-616b8f911a37.png)


Add a label to product excluded to be returned and disable buttons to select quantity and reason.
![2 4-b](https://user-images.githubusercontent.com/96049132/164194184-10a1c147-9f44-4498-8afd-be3cb4cf157e.png)



Connect to GraphQL resolver to get reasons from settings - LOW.
![2 5](https://user-images.githubusercontent.com/96049132/164194209-092ce39d-0eeb-4c97-9144-6e5e7bf2a572.png)
